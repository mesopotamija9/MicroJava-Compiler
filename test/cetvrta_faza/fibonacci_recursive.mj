
program fibonacci_recursive

{
	int fib(int n)
	{
		if (n <= 1)
			return n;
			
		return fib(n - 1) + fib(n - 2);
	}
	
	void main()
		int n;
	{
		n = 9;
		print(fib(n));
	}

}