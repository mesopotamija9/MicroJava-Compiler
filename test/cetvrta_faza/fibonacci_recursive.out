INFO  21:56:06,973 - Compiling source file: D:\Programing Projects\PROG_P1\workspace\MJCompiler\test\cetvrta_faza\fibonacci_recursive.mj
INFO  21:56:07,019 - #17 program
INFO  21:56:07,019 - #2 fibonacci_recursive
INFO  21:56:07,019 - #47 {
INFO  21:56:07,020 - #2 int
INFO  21:56:07,020 - #2 fib
INFO  21:56:07,020 - #43 (
INFO  21:56:07,021 - #2 int
INFO  21:56:07,021 - #2 n
INFO  21:56:07,021 - #44 )
INFO  21:56:07,022 - #47 {
INFO  21:56:07,024 - #23 if
INFO  21:56:07,024 - #43 (
INFO  21:56:07,024 - #2 n
INFO  21:56:07,024 - #11 <=
INFO  21:56:07,026 - #3 1
INFO  21:56:07,026 - #44 )
INFO  21:56:07,028 - #29 return
INFO  21:56:07,028 - #2 n
INFO  21:56:07,029 - #39 ;
INFO  21:56:07,029 - #29 return
INFO  21:56:07,030 - #2 fib
INFO  21:56:07,030 - #43 (
INFO  21:56:07,030 - #2 n
INFO  21:56:07,030 - #13 -
INFO  21:56:07,030 - #3 1
INFO  21:56:07,030 - #44 )
INFO  21:56:07,031 - #12 +
INFO  21:56:07,031 - #2 fib
INFO  21:56:07,031 - #43 (
INFO  21:56:07,031 - #2 n
INFO  21:56:07,031 - #13 -
INFO  21:56:07,031 - #3 2
INFO  21:56:07,031 - #44 )
INFO  21:56:07,032 - #39 ;
INFO  21:56:07,032 - #48 }
INFO  21:56:07,032 - #30 void
INFO  21:56:07,032 - #2 main
INFO  21:56:07,032 - #43 (
INFO  21:56:07,032 - #44 )
INFO  21:56:07,033 - #2 int
INFO  21:56:07,033 - #2 n
INFO  21:56:07,033 - #39 ;
INFO  21:56:07,033 - #47 {
INFO  21:56:07,033 - #2 n
INFO  21:56:07,033 - #36 =
INFO  21:56:07,034 - #3 9
INFO  21:56:07,034 - #39 ;
INFO  21:56:07,034 - #27 print
INFO  21:56:07,034 - #43 (
INFO  21:56:07,035 - #2 fib
INFO  21:56:07,035 - #43 (
INFO  21:56:07,035 - #2 n
INFO  21:56:07,035 - #44 )
INFO  21:56:07,035 - #44 )
INFO  21:56:07,035 - #39 ;
INFO  21:56:07,035 - #48 }
INFO  21:56:07,035 - #48 }
INFO  21:56:07,042 - Program(
  ProgramName(
   fibonacci_recursive
  ) [ProgramName]
  NoDeclList(
  ) [NoDeclList]
  MethodDeclList(
    MethodDeclarationType(
      Type(
       int
      ) [Type]
      MethodNameType(
       fib
      ) [MethodNameType]
      FormParams(
        ParamTypeName(
          Type(
           int
          ) [Type]
         n
        ) [ParamTypeName]
        NoFormParamsMultiple(
        ) [NoFormParamsMultiple]
      ) [FormParams]
      NoVarDeclList(
      ) [NoVarDeclList]
      StatementList(
        StatementList(
          NoStatementList(
          ) [NoStatementList]
          StatementIf(
            IfKeyWord(
            ) [IfKeyWord]
            ConditionCloseParenthesis(
              Condition(
                ConditionTerm(
                  ConditionFactRelopExpr(
                    ExprWithoutMinus(
                      Term(
                        FactorDesignator(
                          Designator(
                            DesignatorIdent(
                             n
                            ) [DesignatorIdent]
                          ) [Designator]
                        ) [FactorDesignator]
                        NoMulopFactor(
                        ) [NoMulopFactor]
                      ) [Term]
                      NoAddopTerminal(
                      ) [NoAddopTerminal]
                    ) [ExprWithoutMinus]
                    RelopLessOrEqual(
                     <=
                    ) [RelopLessOrEqual]
                    ExprWithoutMinus(
                      Term(
                        FactorIntLiteral(
                         1
                        ) [FactorIntLiteral]
                        NoMulopFactor(
                        ) [NoMulopFactor]
                      ) [Term]
                      NoAddopTerminal(
                      ) [NoAddopTerminal]
                    ) [ExprWithoutMinus]
                  ) [ConditionFactRelopExpr]
                  NoConditionFactMultiple(
                  ) [NoConditionFactMultiple]
                ) [ConditionTerm]
                NoConditionTermMultiple(
                ) [NoConditionTermMultiple]
              ) [Condition]
              CondEnd(
              ) [CondEnd]
            ) [ConditionCloseParenthesis]
            StatementReturnExpr(
              ReturnKeyWord(
              ) [ReturnKeyWord]
              ExprWithoutMinus(
                Term(
                  FactorDesignator(
                    Designator(
                      DesignatorIdent(
                       n
                      ) [DesignatorIdent]
                    ) [Designator]
                  ) [FactorDesignator]
                  NoMulopFactor(
                  ) [NoMulopFactor]
                ) [Term]
                NoAddopTerminal(
                ) [NoAddopTerminal]
              ) [ExprWithoutMinus]
            ) [StatementReturnExpr]
            IfThenEnd(
            ) [IfThenEnd]
          ) [StatementIf]
        ) [StatementList]
        StatementReturnExpr(
          ReturnKeyWord(
          ) [ReturnKeyWord]
          ExprWithoutMinus(
            Term(
              FactorFunctionCallWithParams(
                FunctionCallName(
                  Designator(
                    DesignatorIdent(
                     fib
                    ) [DesignatorIdent]
                  ) [Designator]
                ) [FunctionCallName]
                ActualParams(
                  ExprWithoutMinus(
                    Term(
                      FactorDesignator(
                        Designator(
                          DesignatorIdent(
                           n
                          ) [DesignatorIdent]
                        ) [Designator]
                      ) [FactorDesignator]
                      NoMulopFactor(
                      ) [NoMulopFactor]
                    ) [Term]
                    AddopTerminal(
                      NoAddopTerminal(
                      ) [NoAddopTerminal]
                      AddopMinus(
                       -
                      ) [AddopMinus]
                      Term(
                        FactorIntLiteral(
                         1
                        ) [FactorIntLiteral]
                        NoMulopFactor(
                        ) [NoMulopFactor]
                      ) [Term]
                    ) [AddopTerminal]
                  ) [ExprWithoutMinus]
                  NoActualParamsMultiple(
                  ) [NoActualParamsMultiple]
                ) [ActualParams]
              ) [FactorFunctionCallWithParams]
              NoMulopFactor(
              ) [NoMulopFactor]
            ) [Term]
            AddopTerminal(
              NoAddopTerminal(
              ) [NoAddopTerminal]
              AddopPlus(
               +
              ) [AddopPlus]
              Term(
                FactorFunctionCallWithParams(
                  FunctionCallName(
                    Designator(
                      DesignatorIdent(
                       fib
                      ) [DesignatorIdent]
                    ) [Designator]
                  ) [FunctionCallName]
                  ActualParams(
                    ExprWithoutMinus(
                      Term(
                        FactorDesignator(
                          Designator(
                            DesignatorIdent(
                             n
                            ) [DesignatorIdent]
                          ) [Designator]
                        ) [FactorDesignator]
                        NoMulopFactor(
                        ) [NoMulopFactor]
                      ) [Term]
                      AddopTerminal(
                        NoAddopTerminal(
                        ) [NoAddopTerminal]
                        AddopMinus(
                         -
                        ) [AddopMinus]
                        Term(
                          FactorIntLiteral(
                           2
                          ) [FactorIntLiteral]
                          NoMulopFactor(
                          ) [NoMulopFactor]
                        ) [Term]
                      ) [AddopTerminal]
                    ) [ExprWithoutMinus]
                    NoActualParamsMultiple(
                    ) [NoActualParamsMultiple]
                  ) [ActualParams]
                ) [FactorFunctionCallWithParams]
                NoMulopFactor(
                ) [NoMulopFactor]
              ) [Term]
            ) [AddopTerminal]
          ) [ExprWithoutMinus]
        ) [StatementReturnExpr]
      ) [StatementList]
    ) [MethodDeclarationType]
    MethodDeclList(
      MethodDeclarationVoid(
        MethodNameVoid(
         main
        ) [MethodNameVoid]
        NoFormParams(
        ) [NoFormParams]
        VarDeclList(
          NoVarDeclList(
          ) [NoVarDeclList]
          VarDeclaration(
            Type(
             int
            ) [Type]
            VariableName(
             n
            ) [VariableName]
            NoVarDeclMultiple(
            ) [NoVarDeclMultiple]
          ) [VarDeclaration]
        ) [VarDeclList]
        StatementList(
          StatementList(
            NoStatementList(
            ) [NoStatementList]
            StatementDesignatorStatement(
              DesignatorStmtAssign(
                DesignatorStatementAssign(
                  Designator(
                    DesignatorIdent(
                     n
                    ) [DesignatorIdent]
                  ) [Designator]
                  AssignOperator(
                  ) [AssignOperator]
                  ExprWithoutMinus(
                    Term(
                      FactorIntLiteral(
                       9
                      ) [FactorIntLiteral]
                      NoMulopFactor(
                      ) [NoMulopFactor]
                    ) [Term]
                    NoAddopTerminal(
                    ) [NoAddopTerminal]
                  ) [ExprWithoutMinus]
                ) [DesignatorStatementAssign]
              ) [DesignatorStmtAssign]
            ) [StatementDesignatorStatement]
          ) [StatementList]
          StatementPrintWithOneParam(
            ExprWithoutMinus(
              Term(
                FactorFunctionCallWithParams(
                  FunctionCallName(
                    Designator(
                      DesignatorIdent(
                       fib
                      ) [DesignatorIdent]
                    ) [Designator]
                  ) [FunctionCallName]
                  ActualParams(
                    ExprWithoutMinus(
                      Term(
                        FactorDesignator(
                          Designator(
                            DesignatorIdent(
                             n
                            ) [DesignatorIdent]
                          ) [Designator]
                        ) [FactorDesignator]
                        NoMulopFactor(
                        ) [NoMulopFactor]
                      ) [Term]
                      NoAddopTerminal(
                      ) [NoAddopTerminal]
                    ) [ExprWithoutMinus]
                    NoActualParamsMultiple(
                    ) [NoActualParamsMultiple]
                  ) [ActualParams]
                ) [FactorFunctionCallWithParams]
                NoMulopFactor(
                ) [NoMulopFactor]
              ) [Term]
              NoAddopTerminal(
              ) [NoAddopTerminal]
            ) [ExprWithoutMinus]
          ) [StatementPrintWithOneParam]
        ) [StatementList]
      ) [MethodDeclarationVoid]
      NoMethodDeclList(
      ) [NoMethodDeclList]
    ) [MethodDeclList]
  ) [MethodDeclList]
) [Program]
INFO  21:56:07,042 - ===================================
INFO  21:56:07,046 - Obradjuje se deklaracija funkcije fib na liniji 5
INFO  21:56:07,046 - Deklaracija formalnog parametra funkcije -> Var n: int, 0, 1 na liniji 5
INFO  21:56:07,046 - Upotreba simbola 'n' na liniji 7
INFO  21:56:07,046 - Upotreba relazionog operatora: 'int' '<=' 'int' na liniji 7
INFO  21:56:07,046 - Upotreba simbola 'n' na liniji 8
INFO  21:56:07,046 - Upotreba simbola 'fib' na liniji 10
INFO  21:56:07,046 - Upotreba simbola 'n' na liniji 10
INFO  21:56:07,046 - Poziv 'fib()' na liniji 10
INFO  21:56:07,047 - Upotreba simbola 'fib' na liniji 10
INFO  21:56:07,047 - Upotreba simbola 'n' na liniji 10
INFO  21:56:07,047 - Poziv 'fib()' na liniji 10
INFO  21:56:07,047 - Obradjuje se deklaracija funkcije main na liniji 13
INFO  21:56:07,047 - Deklaracija lokalne promenljive -> Var n: int, 0, 1 na liniji 14
INFO  21:56:07,047 - Upotreba simbola 'n' na liniji 16
INFO  21:56:07,047 - Upotreba simbola 'fib' na liniji 17
INFO  21:56:07,047 - Upotreba simbola 'n' na liniji 17
INFO  21:56:07,047 - Poziv 'fib()' na liniji 17
INFO  21:56:07,047 - Poziv 'print()' na liniji 17
INFO  21:56:07,047 - ===================================
=====================SYMBOL TABLE DUMP=========================
Type int: int, -1, -1 
Type char: char, -1, -1 
Con eol: char, 10, 0 
Con null: Class [], 0, 0 
Meth chr: char, 0, 1 
   Var i: int, 0, 1 

Meth ord: int, 0, 1 
   Var ch: char, 0, 1 

Meth len: int, 0, 1 
   Var arr: Arr of notype, 0, 1 

Type bool: bool, -1, -1 
Prog fibonacci_recursive: notype, 0, 1 
   Meth fib: int, 0, 1 
      Var n: int, 0, 1 

   Meth main: notype, 0, 0 
      Var n: int, 0, 1 



INFO  21:56:07,047 - Parsiranje uspesno zavrseno
INFO  21:56:07,052 - Generisanje koda uspesno zavrseno
